### PR TITLE
implement timeout option for resources and updated test cases

### PR DESCRIPTION
**Pull request by hongil0316**
## Description

Supporting timeout option on all resources. 

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

**Comment by gcagle3** [comment]
How are the new timeout options used/called?

File: aws/resources/security_group_types.go [hunk 1]
```diff
@@ -36,6 +36,10 @@ func (sg *SecurityGroup) MaxBatchSize() int {
 	return 50
 }
 
+// func (sg *SecurityGroup) GetAndSetResourceConfig(configObj config.Config) config.ResourceType {
```
Comment (denis256) [inline aws/resources/security_group_types.go:39]: This part is required ?

**Comment by hongil0316** [comment]
> How are the new timeout options used/called?

It's in the README page. https://github.com/gruntwork-io/cloud-nuke/blob/master/README.md#with-timeout.

**Comment by gcagle3** [comment]
> > How are the new timeout options used/called?
> 
> It's in the README page. https://github.com/gruntwork-io/cloud-nuke/blob/master/README.md#with-timeout.

Ah, so cloud-nuke has always supported a timeout, but not all of the resources it worked with actually obeyed/supported the documented timeout?

File: aws/resources/security_group_types.go (hunk 1 above)
Comment (hongil0316) [reply aws/resources/security_group_types.go]: Removed.

File: aws/resources/ec2_subnet_types.go [hunk 2]
```diff
@@ -39,6 +39,10 @@ func (es *EC2Subnet) ResourceIdentifiers() []string {
 	return es.Subnets
 }
 
+// func (es *EC2Subnet) GetAndSetResourceConfig(configObj config.Config) config.ResourceType {
```
[the remark this answers is not in the document]
Comment (hongil0316) [reply aws/resources/ec2_subnet_types.go]: Remove.

**Comment by hongil0316** [comment]
> > > How are the new timeout options used/called?
> > 
> > 
> > It's in the README page. https://github.com/gruntwork-io/cloud-nuke/blob/master/README.md#with-timeout.
> 
> Ah, so cloud-nuke has always supported a timeout, but not all of the resources it worked with actually obeyed/supported the documented timeout?

Yeah so we added support for s3 only because it was the resource that cause timeout issue the most. Just adding support for remaining resources.

**Comment by denis256** [comment]
Conflicting files
README.md

File: aws/resources/rds_global_cluster_types_.go [hunk 3]
```diff
@@ -36,6 +36,10 @@ func (instance *DBGlobalClusters) MaxBatchSize() int {
 	return 49
 }
```
Comment (denis256) [inline aws/resources/rds_global_cluster_types_.go:38]: I suspect the file name shouldn't end with `_`

**Review by gcagle3 — DISMISSED** [review]

**Review by denis256 — APPROVED** [review]

**Comment by hongil0316** [comment]
Moved to https://github.com/gruntwork-io/cloud-nuke/pull/712.

**Pull request closed without merging**